src/middlewared/middlewared/plugins/iscsi.py

```diff
@@ -833,34 +833,22 @@ async def disk_choices(self):
         """
         diskchoices = {}
 
-        zvol_query_filters = [('type', '=', 'DISK')]
-
-        used_zvols = [
-            i['path'] for i in (await self.query(zvol_query_filters))
-        ]
-
         zvols = await self.middleware.call(
-            'pool.dataset.query', [
-                ('type', '=', 'VOLUME'),
-                ('locked', '=', False)
-            ]
-        )
-        zvol_list = []
-        for zvol in zvols:
-            zvol_name = zvol['name']
-            zvol_size = zvol['volsize']['value']
-            zvol_list.append(zvol_name)
-            key = os.path.relpath(zvol_name_to_path(zvol_name), '/dev')
-            if key not in used_zvols:
-                diskchoices[key] = f'{zvol_name} ({zvol_size})'
-
-        zfs_snaps = await self.middleware.call(
-            'zfs.snapshot.query',
-            [['dataset', 'in', [zvol['name'] for zvol in zvols]]],
-            {'select': ['name']},
+            'zfs.dataset.available_zvols_fast',
+            {"additional_information": ["SIZE", "RO", "ATTACHMENT"]}
```

Comment:
Just a nit, but can we use single quotes keeping consistent from above ;) ?

Reply:
Just to be sure, this is not going to get locked zvols ?

Reply:
I tried it and locked volumes don't get presented over /dev/zvol, IMHO it would be a pretty significant bug if they were.